src/diffusers/pipelines/kandinsky/pipeline_kandinsky_prior.py

```diff
@@ -525,9 +556,15 @@ def __call__(
         # if negative prompt has been defined, we retrieve split the image embedding into two
         if negative_prompt is None:
             zero_embeds = self.get_zero_embed(latents.shape[0], device=latents.device)
+
+            if hasattr(self, "final_offload_hook") and self.final_offload_hook is not None:
+                self.final_offload_hook.offload()
         else:
             image_embeddings, zero_embeds = image_embeddings.chunk(2)
 
+            if hasattr(self, "final_offload_hook") and self.final_offload_hook is not None:
+                self.prior_hook.offload()
```

Comment:
```suggestion
            if hasattr(self, "final_offload_hook") and self.final_offload_hook is not None:
                self.final_offload_hook.offload()
```

If I am understanding this correctly we should always offload `final_offload_hook`, so we could put it outside the `if`.

Are we missing a `self.prior_hook.offload()` in `interpolate`?

Reply:
This is actually expected since for this branch the prior is the final model